File: examples/python/bytes_source.py
import mcapable

from common import sample_bytes


def main() -> None:
    data = sample_bytes()
    reader = mcapable.Reader.from_bytes(data)

    header = reader.header()
    print(f"Profile: {header.profile}")
    print(f"Library: {header.library}")

    count = 0
    stream = reader.raw_messages()
    for msg in stream:
        count += 1
        if count <= 3:
            print(
                f"Message: channel={msg.channel_id} time={msg.log_time} size={len(msg.data)}"
            )
    print(f"Total messages: {count}")


if __name__ == "__main__":
    main()

File: examples/typescript/bytes_source.ts
import { Reader } from "@mcapable/wasm";

import { sampleBytes } from "./common";

function main(): void {
  const reader = Reader.fromBytes(sampleBytes());
  const header = reader.header() as { profile: string; library: string };
  console.log(`Profile: ${header.profile}`);
  console.log(`Library: ${header.library}`);

  const stream = reader.raw_messages();
  let count = 0;
  for (;;) {
    const item = stream.next();
    if (!item) break;
    const msg = item as { channel_id: number; log_time: number; data: Uint8Array };
    count += 1;
    if (count <= 3) {
      console.log(`Message: channel=${msg.channel_id} time=${msg.log_time} size=${msg.data.length}`);
    }
  }
  console.log(`Total messages: ${count}`);
}

main();
